tests/infrastructure/textual/widgets/test_agent_workspace_logic.py

```diff
@@ -0,0 +1,49 @@
+
+import pytest
+from pathlib import Path
+from unittest.mock import MagicMock
+from simple_agent.infrastructure.textual.widgets.agent_workspace import AgentWorkspace
+from simple_agent.application.agent_id import AgentId
+from textual.widgets import Markdown
+
+class TestAgentWorkspace:
+    def test_refresh_todos_updates_content_from_file(self, tmp_path):
+        agent_id = AgentId("test_agent")
+        todo_file = tmp_path / "todos.md"
+        todo_file.write_text("Initial content", encoding="utf-8")
+
+        # Override todo_filename to return the absolute path to our temp file
+        agent_id.todo_filename = lambda: str(todo_file)
```

Comment:
We are overriding prod code, this is smelly. Either the prod code is testable, and then we don't need to override it, or we need to first make it testable.